### ISSUE TITLE
proposal - support additional cli args

### ISSUE THREAD

**Issue by travisstaloch** [zigtools/sus#3]
i would like it if the project were more configurable by supporting additional cli args. i think this could be done without introducing a bunch of complexity or dependencies.  

motivation - i think we are likely to discover more bugs in zls/zig if this project is easily configurable.  as an example, i found an adversarial input the other day - `enum(u32)` - after manually changing maxlen and 'cycles per gen' values to smaller numbers.  this happened quickly after changing the parameters while the previous configuration ran for ~15 minutes without discovering any adversarial inputs.

i would be glad to implement this if you agree that its a good idea.  i think it could be done fairly simply with a state-machine (state = mode)  and an args iterator.

my idea for these args would be:
* zls path - positional first arg - full path to the zls executable
* mode - positional second arg - one of [markov, ...]
  * mode == markov
    * --training-dir - directory to be used for training the markov model. searched for .zig files recursively.
    * --maxlen - maximum length of each file generated by the markov model.
    * --cycles-per-gen - how many times to call `fuzzer.fuzzFeatureRandom()` before regenerating a new file
  * mode == ...
    * process other args specific to this mode

let me know if you think this is a good idea and/or have anything to add or change.  no worries if its just too early for these type of changes. just wanted to share my thoughts and maybe get some feedback.

**Comment by SuperAuguste** [comment]
Seems like a good idea in retrospect :)

I think that the main thing is just making all of these default to something so I can just toss the exe onto the server without configuring anything :)

Thanks Travis - feel free to take this on!

**Comment by travisstaloch** [comment]
this is the minimal command for markov. not entirely config free but only 3 required args is pretty short. i'm not sure how t choose defaults for any of the directories but making them default to something would easy.
```console
$ zig build run -- ../zls/zig-out/bin/zls markov ../zig/test/behavior
parsed args: ../zls/zig-out/bin/zls markov ../zig/test/behavior --maxlen 512 --cycles-per-gen 25
...
```
one thing i liked about this pr is it gets rid of a lot of magic values and puts the default values in one place. also it makes args a field of fuzzer so that if more fuzzing modes are added, they can just add a field to the Args.Base union and could share the same api for initialization, with init methods like this:
```zig
    var markov = try Markov.init(markov_arena.allocator(), fuzzer);
```

**Comment by travisstaloch** [comment]
closed by #4
